### ISSUE TITLE
Creating a view in a migration makes problematic schema.rb (MySQL)

### ISSUE THREAD

**Issue by dgsan** [rails/rails#28349]
### Steps to reproduce
Create a migration like so (details don't really matter as long as you make a view):
```ruby
class CreateMyViewView < ActiveRecord::Migration
  def up
    sql_command = %{CREATE VIEW `my_view`  AS
        SELECT `users`.`id` as `id`, `users`.`some_col` as `my_col_name`
        FROM `users` LEFT OUTER JOIN `some_table` ON (`some_table`.`user_id` = `users`.`id`)
        WHERE `users`.`some_col` = "some_value"
        ORDER BY `users`.`another_col` ASC
    } # join some tables and select some stuff to create a view here. Details don't particularly matter.
    execute sql_command
  end
  
  def down
    sql_command = "DROP VIEW `my_view`"
    execute sql_command
  end
```
and run it. (This works.)

Then look at your schema.rb. You'll see something like:
```ruby
create_table "my_view", id: false, force: :cascade do |t|
  t.integer  "id",                               limit: 4,   default: 0, null: false
  # ... and so on ...
end
```
### Expected behavior
Probably whatever makes `schema.rb` should be smart enough to realize that `my_view` isn't a table.

### Actual behavior
The `schema.rb` file "thinks" the view you just created is a table, breaking any attempts to set up your db without using migrations.

### System configuration
**Rails version**: Rails 4.2.7.1

**Ruby version**: ruby 2.3.3p222

**"MySQL" Version**: mysql  Ver 15.1 Distrib 10.1.3-MariaDB

### Notes
Not sure views are commonly used functionality. I tend to use them when data's significantly more complicated than what's needed for display. We've been using one as a way of simplifying read only access to what I'd term "complicated legacy data" that's otherwise next to impossible to work with effectively and is only infrequently accessed by a few internal users for near a year now (with various workarounds). 

Did not know [scenic](https://github.com/thoughtbot/scenic) existed, either, for whatever that's worth. May look into it now that I do know it exists. Not sure its existence negates this as an issue or not.

**Comment by kamipo** [comment]
Since #21609 the `schema.rb` doesn't think (ignores) views.

**Comment by al2o3cr** [comment]
+1 to what @kamipo said - this is fixed in 5.0. Prior to that, you'll need to switch your schema to SQL format (via `config.active_record.schema_format = :sql`) or use something like `scenic` that handles views correctly.
